File: 5/p5.cpp
/*
project euler problem 5
*/

#include <iostream>

bool isDivisible(int num){
  for (size_t i = 11; i < 21; i++) {
    if (num % i != 0) {
      return false;
    }
  }
  return true;
}

int main(int argc, char const *argv[]) {
  int smallest = 20;

  while (!isDivisible(smallest)) {
    smallest += 20;
  }

  std::cout << smallest << '\n';


  return 0;
}

File: 5/p5.py
"""
project euler problem 5
"""
smallest = 20

def is_divisible(num):
    """
    returns true if num is evenly divisible with every number from 1 to 20
    """
    for i in range(11, 21):
        if num % i != 0:
            return False
    return True

while not is_divisible(smallest):
    smallest += 20

print(smallest)
